File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02891/p02891_num9_parsed.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string S;
    int K;

    // Read input
    cin >> S >> K;

    // Count the number of adjacent equal characters in S
    int count = 0;
    for (int i = 1; i < S.length(); i++) {
        if (S[i] == S[i - 1]) {
            count++;
        }
    }

    // Calculate the minimum number of operations required
    int operations = 0;
    if (count > 0) {
        operations = (count + 1) * K / 2;
    }

    // Output the result
    cout << operations << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02891/p02891_num9_parsed.py
def min_operations(S, K):
    length = len(S)
    count = 0
    
    # If K is 1, then no operations needed
    if K == 1:
        return count
    
    # Check if all characters in S are the same
    if all(x == S[0] for x in S):
        # If all characters are the same, then only need to change the first character of each copy of S in T
        return length // 2
    
    # Count the number of character pairs that need to be changed
    for i in range(1, length):
        if S[i-1] == S[i]:
            count += 1

    # Calculate the number of additional changes needed to make all adjacent characters different
    additional_changes = (K - 1) * count
    
    # Return the total number of operations needed
    return count + additional_changes

S = input()
K = int(input())

result = min_operations(S, K)
print(result)